python-avd/pyavd/_anta/logs.py

```
@@ -49,20 +52,27 @@ def process(self, msg: LogMessage, kwargs: dict) -> tuple[str, dict]:
 
         # Format the LogMessage using the provided kwargs and extract the fields name from the message string
         fields = [field_name for _, field_name, _, _ in string.Formatter().parse(msg.value) if field_name is not None]
-        msg = msg.value.format(**kwargs)
+        processed_msg = msg.value.format(**kwargs)
 
         # Removing the fields name from kwargs to preserve standard logging kwargs only that should always be passed through (e.g. exc_info, stack_info, etc.)
         for field in fields:
             kwargs.pop(field, None)
 
-        return f"{prefix} {msg}", kwargs
+        return f"{prefix} {processed_msg}", kwargs
 
     @contextmanager
     def context(self, context: str) -> Generator[TestLoggerAdapter, None, None]:
         """Temporarily add context to the logger."""
-        original_extra = dict(self.extra)
+        if self.extra is None:
```

Since our custom adapter relies on extra **not** being None, we should probably override `__init__` and raise if extra is not provided with device and test. We would have the proper type everywhere and we wouldn't need all that additional complexity.